pymediainfo/__init__.py

```diff
@@ -346,11 +351,23 @@ def parse(cls, filename, library_file=None, cover_data=False,
                 )
             for option_name, option_value in mediainfo_options.items():
                 lib.MediaInfo_Option(handle, option_name, option_value)
-        if lib.MediaInfo_Open(handle, filename) == 0:
-            lib.MediaInfo_Close(handle)
-            lib.MediaInfo_Delete(handle)
-            raise RuntimeError("An error occured while opening {}"
-                    " with libmediainfo".format(filename))
+
+        try:
+            file_bytes = filename.read()
```

Comment:
This is the thing that bothers me most. Isn't there a way to pass the file descriptor to the underlying library?

It would be nicer than reading the whole file into memory.

If not, the documentation should mention performance limitations of file-like objects compared to files/URLs.

Reply:
> Isn't there a way to pass the file descriptor to the underlying library?

@JeromeMartinez should know better than me, if so I missed it.

Reply:
> It would be nicer than reading the whole file into memory.

Yes. I'm working on jpgs currently (videos will come later) so I easily forgot that... I'll read the next comment, there may be a better way to use the buffer API.

Reply:
File-like objects seem very Python, not C/C++, so the best way is to use the "by buffer" API, also for images (no need to read the whole file also for small files).
In practice it is the API used internally.

But with the full example implemented, so managing the seek requests, and sending e.g. 64KiB by 64 KiB (hum... was my choice 15 years ago, maybe another value is more optimal nowadays)

Reply:
To avoid loading the whole file in memory, what about something like:

```
with open(filename, "rb") as f:
    with mmap.mmap(f.fileno(), 0, prot=mmap.PROT_READ) as mm:
        lib.MediaInfo_Open_Buffer(handle, bytes(mm), len(mm), b"", 0)
        info = lib.MediaInfo_Inform(handle, 0)
```

the `bytes(mm)` part should not copy it (hum, maybe in 2.7) it should use the buffer protocol to avoid the copy.

Reply:
Can't we try to use the functions from the example Jérôme linked? I won't have time to look at this in great depth before Monday at the earliest.